reactor-core/src/main/java/reactor/core/publisher/MonoCacheTime.java

```diff
@@ -287,12 +287,15 @@ private void signalCached(Signal<T> signal) {
 					}
 				}
 
-				if (ttl != null && ttl.isZero()) {
-					//immediate cache clear
-					main.run();
-				}
-				else if (ttl != null) {
-					main.clock.schedule(main, ttl.toMillis(), TimeUnit.MILLISECONDS);
+				if (ttl != null) {
+					if (ttl.isZero()) {
+						//immediate cache clear
+						main.run();
+					}
+					else if (ttl.toMillis() < Long.MAX_VALUE) {
```

Comment:
`ttl.toMillis()` is calculated two times, maybe it is not a big issue but it could be avoided if make smth like 
```java
else {
    long millis = ttl.toMillis();
    if (millis < Long.MAX_VALUE) {
        main.clock.schedule(main, millis, TimeUnit.MILLISECONDS);
    }
    //else TTL is Long.MAX_VALUE, schedule nothing but still cache
}
```

Reply:
good suggestion. I could also precompute `Duration.ofMillis(Long.MAX_VALUE)` in a constant and do a direct equals comparison with the `ttl` 🤔

Reply:
done ☝️